coderd/workspaces.go

```diff
@@ -952,6 +953,29 @@ func (api *API) putWorkspaceDormant(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// We don't need to notify the owner if they are the one making the request.
+	if req.Dormant && apiKey.UserID != workspace.OwnerID {
+		initiator, err := api.Database.GetUserByID(ctx, apiKey.UserID)
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "Internal error fetching user.",
+				Detail:  err.Error(),
+			})
+			return
+		}
+		dormancy.NotifyWorkspaceDormant(
+			ctx,
+			api.Logger,
+			api.NotificationsEnqueuer,
+			dormancy.WorkspaceDormantNotification{
+				Workspace:   workspace,
+				InitiatedBy: initiator.Username,
+				Reason:      "requested by user",
```

Comment:
Is this API only ever called by user-initiated actions? I had a brief look and it seems to be, but maybe you have more context. If that's the case, we don't need this constant value - we can just always say "requested by user" in the template.

Reply:
I need to implement this notification in two places: the lifecycle executor and the scheduler.